docs/compilers/CSharp/Compiler Breaking Changes - DotNet 8.md

```diff
@@ -0,0 +1,28 @@
+# This document lists known breaking changes in Roslyn after .NET 7 all the way to .NET 8.
```

Comment:
Thanks for adding the doc. Once the question of whether the PR is ready to merge, please email a copy of the added section to "Roslyn Compatibility Council" and LDM aliases as FYI.